src/Maker/AdminMaker.php

```diff
@@ -0,0 +1,265 @@
+            ->addOption('id', 'i', InputOption::VALUE_OPTIONAL, 'The admin service ID');
+
+        $inputConfig->setArgumentAsNonInteractive('model');
+    }
+
+    public function interact(InputInterface $input, ConsoleStyle $io, Command $command)
+    {
+        $io->section('Welcome to the Sonata Admin');
+        $modelClass = $io->ask(
+            'The fully qualified model class',
+            $input->getArgument('model'),
+            [Validators::class, 'validateClass']
+        );
+        $modelClassBasename = current(\array_slice(explode('\\', $modelClass), -1));
+
+        $adminClassBasename = $io->ask(
+            'The admin class basename',
+            $input->getOption('admin') ?: $modelClassBasename.'Admin',
+            [Validators::class, 'validateAdminClassBasename']
+        );
+        if (\count($this->availableModelManagers) > 1) {
+            $managerTypes = array_keys($this->availableModelManagers);
+            $managerType = $io->choice('The manager type', $managerTypes, $managerTypes[0]);
+
+            $input->setOption('manager', $managerType);
+        }
+        if ($io->confirm('Do you want to generate a controller?', false)) {
+            $controllerClassBasename = $io->ask(
+                'The controller class basename',
+                $input->getOption('controller') ?: $modelClassBasename.'AdminController',
+                [Validators::class, 'validateControllerClassBasename']
+            );
+            $input->setOption('controller', $controllerClassBasename);
+        }
+        $input->setOption('services', false);
+        if ($io->confirm('Do you want to update the services YAML configuration file?', true)) {
+            $path = $this->projectDirectory.'/config/';
+            $servicesFile = $io->ask(
+                'The services YAML configuration file',
+                is_file($path.'admin.yaml') ? 'admin.yaml' : 'services.yaml',
+                [Validators::class, 'validateServicesFile']
+            );
+            $id = $io->ask(
+                'The admin service ID',
+                $this->getAdminServiceId($adminClassBasename),
+                [Validators::class, 'validateServiceId']
+            );
+            $input->setOption('services', $servicesFile);
+            $input->setOption('id', $id);
+        }
+        $input->setArgument('model', $modelClass);
+        $input->setOption('admin', $adminClassBasename);
+    }
+
+    /**
+     * Configure any library dependencies that your maker requires.
+     */
+    public function configureDependencies(DependencyBuilder $dependencies)
+    {
+        // TODO: Implement configureDependencies() method.
+    }
+
+    /**
+     * Called after normal code generation: allows you to do anything.
+     */
+    public function generate(InputInterface $input, ConsoleStyle $io, Generator $generator)
+    {
+        $this->configure($input);
+
+        $adminClassFullName = $this->generateAdmin($io, $generator);
```

Comment:
Name of var is `$adminClassFullName` and the name of the method is `generateAdmin` from method name I would expect an instance of admin class, I would say we need a better name for the method.

Reply:
Or maybe segregate the method and have like `generateAdminClassName` and then pass that to `generateAdmin`